### PR TITLE
weightedtarget: handle updating child policy name

### PR DESCRIPTION
**Pull request by menghanl**
Also remove dependency on cds from the tests, otherwise later it will cause cycular dependency: weightedtarget -> cds -> eds -> weightedtarget

### CODE REVIEW

File: xds/internal/balancer/weightedtarget/weightedtarget.go [hunk 1]
```diff
@@ -115,6 +115,16 @@ func (w *weightedTargetBalancer) UpdateClientConnState(s balancer.ClientConnStat
 			w.bg.Add(name, balancer.Get(newT.ChildPolicy.Name))
 			// Not trigger a state/picker update. Wait for the new sub-balancer
 			// to send its updates.
+		} else if newT.ChildPolicy.Name != oldT.ChildPolicy.Name {
```
Comment (easwars) [inline xds/internal/balancer/weightedtarget/weightedtarget.go:118]: What if the child policy name hasn't changed, but the config has changed? Is that possible? Do we need to handle that?

Comment (menghanl) [reply xds/internal/balancer/weightedtarget/weightedtarget.go:118]: That's handled later. We will forward the config and addresses to the children, `w.bg.UpdateClientConnState()`

File: xds/internal/balancer/weightedtarget/weightedtarget_config_test.go [hunk 2]
```diff
@@ -24,32 +24,32 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"google.golang.org/grpc/balancer"
 	internalserviceconfig "google.golang.org/grpc/internal/serviceconfig"
-	_ "google.golang.org/grpc/xds/internal/balancer/cdsbalancer"
+
+	_ "google.golang.org/grpc/xds/internal/balancer/lrs"
```
Comment (easwars) [inline xds/internal/balancer/weightedtarget/weightedtarget_config_test.go:28]: Sorry didn't notice last time around. A small comment explaining why the blank import is done.

**Review by easwars — APPROVED** [review]

File: xds/internal/balancer/weightedtarget/weightedtarget_config_test.go (hunk 2 above)
Comment (menghanl) [reply xds/internal/balancer/weightedtarget/weightedtarget_config_test.go]: Done

**Pull request merged**